### PR TITLE
Add a check that simd is runnable.

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

A very basic check that just calling `simd` doesn't result in any runtime errors, a sanity check before pushing to ghcr.

closes: #3852 

### Commit Message / Changelog Entry

```bash
chore: add a check that simd is runnable before pushing to ghcr.
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -21,19 +21,30 @@ jobs:
     - name: Checkout
       uses: actions/checkout@v3
 
+    - name: Extract metadata (tags, labels) for Docker
+      id: meta
+      uses: docker/metadata-action@818d4b7b91585d195f67373fd9cb0332e31a7175
+      with:
+        images: ${{ env.REGISTRY }}/cosmos/${{ env.IMAGE_NAME }}
+
+    - name: Build and push Docker image
```
Comment (chatton) [inline .github/workflows/docker.yml:30]: this task isn't pushing anymore right? We can update the name

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -21,19 +21,30 @@ jobs:
...
+      uses: docker/build-push-action@2eb1c1961a95fc15694676618e422e8ba1d63825
+      with:
+        context: .
+        tags: ${{ steps.meta.outputs.tags }}
+        build-args: |
```
Comment (chatton) [inline .github/workflows/docker.yml:35]: if neither `push` or `load` are specified, what exactly does this do?

**Review by chatton — COMMENTED** [review]
Left a few small comments trying to understand the new flow! Thanks for creating this PR ❤️

File: .github/workflows/docker.yml [hunk 3]
```diff
@@ -21,19 +21,30 @@ jobs:
...
         username: ${{ github.actor }}
         password: ${{ secrets.GITHUB_TOKEN }}
 
-    - name: Extract metadata (tags, labels) for Docker
-      id: meta
-      uses: docker/metadata-action@818d4b7b91585d195f67373fd9cb0332e31a7175
-      with:
-        images: ${{ env.REGISTRY }}/cosmos/${{ env.IMAGE_NAME }}
-
     - name: Build and push Docker image
```
[the remark this answers is not in the document]
Comment (DimitrisJim) [reply .github/workflows/docker.yml:48]: I tested this in my fork of ibc-go and it did not [don't know if logs are visible to you](https://github.com/DimitrisJim/ibc-go/actions/runs/5296535670/jobs/9587707581#step:7:150).

File: .github/workflows/docker.yml (hunk 1 above)
Comment (DimitrisJim) [reply .github/workflows/docker.yml:30]: yeah, good point!

File: .github/workflows/docker.yml (hunk 2 above)
Comment (DimitrisJim) [reply .github/workflows/docker.yml:35]: just the build. Unsure if there's a better way to do this (combining this step with `test simd is runnable`?)

Comment (DimitrisJim) [reply .github/workflows/docker.yml:35]: ah I guess it doesn't export it to docker? As far as I could tell, this did use the correct image [with the run I triggered on my fork](https://github.com/DimitrisJim/ibc-go/actions/runs/5296535670/jobs/9587707581#step:5:1). Lmk if I'm missing something here.

File: .github/workflows/docker.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (chatton) [reply .github/workflows/docker.yml:48]: ah excellent yeah it just used the cached layers, sounds good to me!

**Review by damiannolan — APPROVED** [review]

**Pull request merged**